src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -403,6 +410,30 @@ public function getDefinedOptions(): array
         return array_keys($this->defined);
     }
 
+    /**
+     * Defines nested options.
+     *
+     * @param \Closure(self $resolver, Options $parent): void $nested
+     *
+     * @return $this
+     */
+    public function setOptions(string $option, \Closure $nested): static
+    {
+        if ($this->locked) {
+            throw new AccessException('Nested options cannot be defined from a lazy option or normalizer.');
+        }
+
+        // Store closure for later evaluation
+        $this->nested[$option][] = $nested;
+        $this->defaults[$option] = [];
+        $this->defined[$option] = true;
+
+        // Make sure the option is processed
+        unset($this->resolved[$option]);
+
+        return $this;
+    }
+
     public function isNested(string $option): bool
     {
         return isset($this->nested[$option]);
```

Comment:
just wondering is this method needs an update too?

Reply:
for the purpose of this feature, nope. or are you asking for a renaming of the method?

Reply:
for the purpose of this feature, so all is fine, thanks for your answer